### DISCUSSION TITLE
Possible to use Roles `IP access` to prevent users from downloading fonts?

### DISCUSSION THREAD

**Discussion by ghost** [directus/directus#17297] [Archive]
I was wondering if it is possible to use Roles’ `IP access` to prevent users from downloading fonts?

I’ve uploaded font files to a folder called `Fonts`. I created a `Webfonts` role, that (is the only role) that can access this folder. I’ve also set the `IP access` to my **local server IP**.

What I’ve expected is that only the website displays the fonts, but no user is able to download the font files via the browser.

So in short: I am trying to prevent a user from downloading files directly through User Roles’ IP access list.

Is it possible with `IP access`?

**Comment by rijkvanzanten** [comment]
> What I’ve expected is that only the website displays the fonts, but no user is able to download the font files via the browser.

This is an interesting use case! When a website displays a font, the user's browser has to download th font file first, so wouldn't this in turn mean this is a bit of an impossibility? 🤔

**Reply by ghost** [reply]
@rijkvanzanten 
When buying a web font you often agree to terms where you have to protect the font files from being directly downloaded via the browser (through the address bar).

Here is an example snippet for a `.htaccess` file that works (on apache):

```htaccess
SetEnvIf Referer example\.com localreferer

<FilesMatch \.(woff|woff2)$>
Order deny,allow
Deny from all
Allow from env=localreferer

FileETag None
<ifModule mod_headers.c>
Header unset ETag
Header set Cache-Control "max-age=0, no-cache, no-store, must-revalidate"
Header set Pragma "no-cache"
Header set Expires "Wed, 11 Jan 1984 05:00:00 GMT"
</ifModule>
</FilesMatch>
```

This code prevents users from downloading files via the address bar. It is only possible via `example.com`.

What I want to achieve, is to load the files from the directus files volume, so that the font files are not part of any git repository.

**Reply by paescuj** [reply]
This isn't an IP restriction per se, but limits access to requests with a [Referer](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Referer) pointing to your domain. This means the files are still transferred to the client when opening the page...

Do you want to use those fonts inside Directus?

**Reply by rijkvanzanten** [reply]
(Also you could theoretically spoof the `Referer` header if you're making the request from somewhere outside of the browser, but I digress 😁 )

This discussion makes me think of a bigger improvement though; it'd be nice if you could include parts of the current request in the permissions rules in some sort of conditional set, so you could say like "If request header referer starts with example.com, allow read access" 🤔

**Reply by paescuj** [reply]
> (Also you could theoretically spoof the `Referer` header if you're making the request from somewhere outside of the browser, but I digress 😁 )

Very likely 😁

> This discussion makes me think of a bigger improvement though; it'd be nice if you could include parts of the current request in the permissions rules in some sort of conditional set, so you could say like "If request header referer starts with example.com, allow read access" 🤔

Wow, interesting! 🤩

**Reply by ghost** [reply]
> This discussion makes me think of a bigger improvement though; it'd be nice if you could include parts of the current request in the permissions rules in some sort of conditional set, so you could say like "If request header referer starts with example.com, allow read access" 🤔 

Well, this is exactly what I was looking for.

Obviously it does not work with your existing `IP access` restriction.

**Reply by ghost** [reply]
> This isn't an IP restriction per se, but limits access to request with a [Referer](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Referer) pointing to your domain. This means the files are still transferred to the client when opening the page...

Of course. I didn't claim otherwise 😁
